src/main.rs

```diff
@@ -33,7 +33,12 @@ pub fn benchmark_smt() {
         entries.push((key, value));
     }
 
-    let mut tree = construction(entries, tree_size).unwrap();
+    let mut tree = construction(entries.clone(), tree_size).unwrap();
+    #[cfg(feature = "concurrent")]
+    {
+        let parallel = parallel_construction(entries, tree_size).unwrap();
+        assert_eq!(tree, parallel);
+    }
```

Comment:
Do we still need this? Wouldn't compiling with or without the `concurrent` feature be sufficient?